frontend/src/features/shoplist/ShoplistPreview.tsx

```
@@ -64,15 +54,43 @@ export default function ShoplistPreview() {
+          URL.revokeObjectURL(url);
+        }
+      } catch {
+        // Ignore errors if URL was already revoked or invalid
+      }
+    }, 10000); // 10 seconds for very large files on slow networks
+
+    // Clear timeouts if component unmounts
+    return () => {
+      clearTimeout(removeAnchorTimeout);
+      clearTimeout(revokeTimeout);
+    };
```

Biome (2.1.2)

[error] Unsafe usage of 'return'.

'return' in 'finally' overwrites the control flow statements inside 'try' and 'catch'.

(lint/correctness/noUnsafeFinally)